library/Zend/InputFilter/Input.php

```diff
@@ -326,6 +326,10 @@ public function isValid($context = null)
             return true;
         }
 
+        if ($empty && !$this->isRequired()) {
```

Comment:
<code>!$this->isRequired()</code> can be change with <code>$this->isRequired() === false</code> imo